pygmt/helpers/tempfile.py

```diff
@@ -104,3 +105,54 @@ def loadtxt(self, **kwargs):
             Data read from the text file.
         """
         return np.loadtxt(self.name, **kwargs)
+
+
+@contextmanager
+def tempfile_from_geojson(geojson):
+    """
+    Saves any geo-like Python object which implements ``__geo_interface__``
+    (e.g. a geopandas GeoDataFrame) to a temporary OGR_GMT text file.
+
+    Parameters
+    ----------
+    geojson : geopandas.GeoDataFrame
+        A geopandas GeoDataFrame, or any geo-like Python object which
+        implements __geo_interface__, i.e. a GeoJSON
```

Comment:
```suggestion
        implements __geo_interface__, i.e. a GeoJSON.
```

Reply:
Looks great to me @weiji14. Except the missing period I have no proposed changes at the moment. What are the open points for this PR?

Reply:
See my comments at https://github.com/GenericMappingTools/pygmt/pull/1000#discussion_r587969541 and https://github.com/GenericMappingTools/pygmt/pull/1000#discussion_r606965331. I'll try and tackle the former docstring/intersphinx issue, but would appreciate some help on the latter shapely/fiona problem.